src/views/Products.jsx

```diff
@@ -1,7 +1,104 @@
-import React from 'react'
+/* eslint-disable no-unused-vars */
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import ProductsSidebar from '../components/ProductsSidebar'
+import ProductCategories from '../mocks/en-us/product-categories.json'
+import ProductsJson from '../mocks/en-us/products.json'
+import '../styles/Products.scss'
+import ProductItem from '../components/FeaturedProducts/ProductItem'
+import ProductsPagination from '../components/ProductsPagination'
+import Spinner from '../components/Spinner'
+
+export default function Products ({ onHandleLinkPage }) {
+  const [isLoading, setIsLoading] = useState(true)
+  const [filters, setFilters] = useState([])
+  const [filteredProducts, setFilteredProducts] = useState(ProductsJson.results)
+
+  const isInFilters = (value) => {
+    return filters.indexOf(value) !== -1
+  }
+
+  const filterProducts = () => {
+    const products = ProductsJson.results.filter((fProduct) => {
+      return filters.includes(fProduct.data.category.id)
+    })
+    setFilteredProducts(products)
+  }
+
+  const resetLoading = () => {
+    setTimeout(() => {
+      setIsLoading(false)
+    }, 2000)
+  }
+
+  const handleFilters = (e) => {
+    const { checked, value } = e.target
+    setIsLoading(true)
+
+    if (checked) {
+      filters.push(value)
+      setFilters(filters)
+    } else if (isInFilters(value)) {
+      filters.splice(filters.indexOf(value), 1)
+      setFilters(filters)
+    }
```

Comment:
the `if/else` creates a bifurcation of the program logic, in natural language days something like

> if checked push the value to filters and then update the filters, else remove the value from the filters and update the filters

that could be simplified because at the end you will always update the filters, the logic can be changed to:

> if checked push the value else remove the value, then update the filters

Another thing is the mutation of the variable `filters`, in react it is preferred to never mutate variables to prevent unexpected behaviors, [this article](https://blog.logrocket.com/immutability-in-react-ebe55253a1cc/) explains more about the subject

An alternative to the code could be something like:

```jsx
const newFilters = checked
  ? [...filters, value]
  : filters.filter(i => i !== value)
setFilters(newFilters)
```